### PR TITLE
Update core library to 6.4

### PR DESCRIPTION
**Pull request by henryju**
I adapted HTTP client implementation from SLI, so maybe @damien-urruty-sonarsource you might be interested to give a look?

### CODE REVIEW

File: src/main/java/org/sonarsource/sonarlint/ls/AnalysisManager.java [hunk 1]
```diff
@@ -133,7 +133,7 @@
   private final ProjectBindingManager bindingManager;
   private final EventWatcher watcher;
   private final LanguageClientLogOutput lsLogOutput;
-  private final Map<String, Boolean> filesIgnoredByScmCache = new HashMap<>();
+  private final ScmIgnoredCache filesIgnoredByScmCache;
```
Comment (jblievremont) [inline src/main/java/org/sonarsource/sonarlint/ls/AnalysisManager.java:136]: Nothing to do with this commit, but I believe that this cache can become stale when the `.gitignore` is modified (e.g a file that was not ignored can become ignored, and the cache won't know it until the user closes/reopen the file).

File: src/main/java/org/sonarsource/sonarlint/ls/AnalysisManager.java [hunk 2]
```diff
@@ -275,7 +276,7 @@ public void didClose(URI fileUri) {
     taintVulnerabilitiesPerFile.remove(fileUri);
     eventMap.remove(fileUri);
     client.publishDiagnostics(newPublishDiagnostics(fileUri));
-    filesIgnoredByScmCache.clear();
+    filesIgnoredByScmCache.remove(fileUri);
```
Comment (jblievremont) [inline src/main/java/org/sonarsource/sonarlint/ls/AnalysisManager.java:279]: 👍🏻

**Review by jblievremont — APPROVED** [review]
LGTM 👍🏻

File: src/main/java/org/sonarsource/sonarlint/ls/AnalysisManager.java (hunk 1 above)
Comment (henryju) [reply src/main/java/org/sonarsource/sonarlint/ls/AnalysisManager.java:136]: Yes, this is what I suggested to Kirill to be an acceptable limitation.

**Pull request merged**